base/intfuncs.jl

```diff
@@ -174,8 +173,8 @@ function power_by_squaring(x_, p::Integer)
     elseif p == 2
         return x*x
     elseif p < 0
-        x == 1 && return copy(x)
-        x == -1 && return iseven(p) ? one(x) : copy(x)
+        isone(x) && return copy(x)
+        isone(-x) && return iseven(p) ? one(x) : copy(x)
         throw_domerr_powbysq(p)
```

Comment:
I would put it here, since this is where it is used. If we add a specialized power by squaring method for matrices it can be moved to `dense.jl` later on. (Which, btw would be a good idea, since we can probably allocate some work arrays and allocate less in matrix multiplication, but thats for another PR :)).